public/components/preview_panel/index.tsx

```diff
@@ -31,7 +31,11 @@ export interface PreviewModel {
   name: string;
   id: string;
   planningWorkerNodes: string[];
-  source: string;
+  connector?: {
+    id?: string;
+    name?: string;
+    description?: string;
+  };
```

Comment:
I think the paras passed should be 
`
  connector?: {
    id?: string;
    name: string;
    description?: string;
  };
` 
Or in the case of external, the name may also be empty?

Reply:
Sure, let me check if we can modify this name to required.